File: gitdot-web/packages/api/src/endpoint/review/remove-reviewer.ts
import { z } from "zod";
import type { Endpoint } from "../endpoint";

export const RemoveReviewerRequest = z.object({});
export type RemoveReviewerRequest = z.infer<typeof RemoveReviewerRequest>;

export const RemoveReviewerResponse = z.void();
export type RemoveReviewerResponse = z.infer<typeof RemoveReviewerResponse>;

export const RemoveReviewer = {
  path: "/repository/{owner}/{repo}/review/{number}/reviewer/{reviewer_name}",
  method: "DELETE",
  request: RemoveReviewerRequest,
  response: RemoveReviewerResponse,
} as const satisfies Endpoint;
export type RemoveReviewer = typeof RemoveReviewer;

File: gitdot-api/src/endpoint/review/remove_reviewer.rs
use serde::{Deserialize, Serialize};

use crate::endpoint::Endpoint;

pub struct RemoveReviewer;

impl Endpoint for RemoveReviewer {
    const PATH: &'static str =
        "/repository/{owner}/{repo}/review/{number}/reviewer/{reviewer_name}";
    const METHOD: http::Method = http::Method::DELETE;

    type Request = RemoveReviewerRequest;
    type Response = ();
}

#[derive(ApiRequest, Debug, Serialize, Deserialize)]
pub struct RemoveReviewerRequest;
